test/index.js

```diff
@@ -0,0 +1,186 @@
+      'repositoryDir':    'pages-repos',
+      'generatedSiteDir': 'pages-generated',
+      'secretKeyFile':    path.join(__dirname, 'data', '18f-pages.secret'),
+    }
+  ]
+};
+
+function makeSignature(payload, secret) {
+  return 'sha1=' +
+    crypto.createHmac('sha1', secret).update(payload).digest('hex');
+}
+
+function httpOptions(port, payload, secret) {
+  return {
+    hostname: 'localhost',
+    port: port,
+    path: '/',
+    method: 'POST',
+    headers: {
+      'Request URL': 'https://pages.18f.gov/deploy',
+      'Request method': 'POST',
+      'content-type': 'application/json',
+      'Expect': '',
+      'User-Agent': 'GitHub-Hookshot/9db916b',
+      'X-GitHub-Delivery': '01234567-0123-0123-1234-0123456789ab',
+      'X-GitHub-Event': 'push',
+      'X-Hub-Signature': makeSignature(payload, secret),
+      'Content-Type': 'application/json',
+      'Content-Length': payload.length
+    }
+  };
+}
+
+function makePayload(branch) {
+  return JSON.stringify({
+    'ref': 'refs/heads/' + branch,
+    'repository': {
+      'name': 'foo',
+      'full_name': '18F/foo',
+      // Alter the name of the organization so the request is silently ignored
+      // to avoid tons of server logging in the background.
+      'organization': '19G'
+    },
+    'head_commit': {
+      'id': 'deadbeef',
+      'message': 'Build me',
+      'timestamp': '2015-09-25',
+      'committer': { 'email': 'michael.bland@gsa.gov' }
+    },
+    'pusher': { 'name': 'Mike Bland', 'email': 'michael.bland@gsa.gov' },
+    'sender': { 'login': 'mbland' }
+  });
+}
+
+function sendRequest(options, payload) {
+  return new Promise(function(resolve, reject) {
+    var req = http.request(options, function(res) {
+      var data = '';
+      res.setEncoding('utf8');
+      res.on('data', function(chunk) { data += chunk; });
+      res.on('end', function() {
+        if (res.statusCode >= 200 && res.statusCode <= 300) {
+          resolve(data);
+        } else {
+          reject(data);
+        }
+      });
+    });
+
+    req.on('error', function(e) { reject(e.message); });
+    req.write(payload);
+    req.end();
+  });
+}
+
+describe('launchServer', function() {
+  var server, port;
+  var origSpawn, mySpawn;
+  var defaultKey, pagesBranchKey;
+
+  var captureLogs = function() {
+    sinon.stub(console, 'log').returns(null);
+    sinon.stub(console, 'error').returns(null);
+  };
+
+  var restoreLogs = function() {
+    console.error.restore();
+    console.log.restore();
+  };
+
+  before(function(done) {
+    defaultKey = fs.readFileSync(config.secretKeyFile, 'utf8').trim();
+    pagesBranchKey = fs.readFileSync(
+      config.builders[0].secretKeyFile, 'utf8').trim();
+    captureLogs();
+    pagesServer.launchServer(config).then(function(runningServer) {
+      server = runningServer;
+      port = server.address().port;
+      restoreLogs();
+      done();
+    }).catch(function(err) { restoreLogs(); done(err); });
```

Comment:
Hmm, I don't know if it could help with the setup... Maybe with sending the request, though I'm not 100% certain. Definitely going to study that framework some more, though; never heard of it before.